diff --git a/sdk/objc/components/audio/RTCAudioSessionConfiguration.m b/sdk/objc/components/audio/RTCAudioSessionConfiguration.m
index c583868291d45ecba9ad3f58087d0eeeedcec095..1feac2df59b3fb7f462fdf79b9f7648809a68ed1 100644
--- a/sdk/objc/components/audio/RTCAudioSessionConfiguration.m
+++ b/sdk/objc/components/audio/RTCAudioSessionConfiguration.m
@@ -60,11 +60,13 @@ static RTC_OBJC_TYPE(RTCAudioSessionConfiguration) *gWebRTCConfiguration = nil;
     // nonmixable, hence activating the session will interrupt any other
     // audio sessions which are also nonmixable.
     _category = AVAudioSessionCategoryPlayAndRecord;
+#if !TARGET_OS_TV
 #if defined(__IPHONE_26_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_26_0
     _categoryOptions = AVAudioSessionCategoryOptionAllowBluetoothHFP;
 #else
     // Use the deprecated option on older SDKs.
     _categoryOptions = AVAudioSessionCategoryOptionAllowBluetooth;
+#endif
 #endif
 
     // Specify mode for two-way voice communication (e.g. VoIP).
diff --git a/sdk/objc/components/capturer/RTCCameraVideoCapturer.h b/sdk/objc/components/capturer/RTCCameraVideoCapturer.h
index 5185fa866e5147c3acee7f27f95143a505505d3f..d5297983ba7a85bdfd6cbb0bd1c1e08e39384f5d 100644
--- a/sdk/objc/components/capturer/RTCCameraVideoCapturer.h
+++ b/sdk/objc/components/capturer/RTCCameraVideoCapturer.h
@@ -19,6 +19,7 @@ NS_ASSUME_NONNULL_BEGIN
 // Camera capture that implements RTCVideoCapturer. Delivers frames to a
 // RTCVideoCapturerDelegate (usually RTCVideoSource).
 NS_EXTENSION_UNAVAILABLE_IOS("Camera not available in app extensions.")
+API_AVAILABLE(tvos(17.0), ios(12.0), macos(10.11))
 RTC_OBJC_EXPORT
 @interface RTC_OBJC_TYPE (RTCCameraVideoCapturer) : RTC_OBJC_TYPE(RTCVideoCapturer)
 
diff --git a/sdk/objc/components/capturer/RTCCameraVideoCapturer.m b/sdk/objc/components/capturer/RTCCameraVideoCapturer.m
index ff55bd7f194fe2c604517b3e3860d0a1326b861f..bd4a0e2338c5e3222362f30aa04dadaab0e5257d 100644
--- a/sdk/objc/components/capturer/RTCCameraVideoCapturer.m
+++ b/sdk/objc/components/capturer/RTCCameraVideoCapturer.m
@@ -15,7 +15,7 @@
 #import "base/RTCVideoFrameBuffer.h"
 #import "components/video_frame_buffer/RTCCVPixelBuffer.h"
 
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
 #import "helpers/UIDevice+RTCDevice.h"
 #endif
 
@@ -42,7 +42,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
   FourCharCode _preferredOutputPixelFormat;
   FourCharCode _outputPixelFormat;
   RTCVideoRotation _rotation;
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
   UIDeviceOrientation _orientation;
   BOOL _generatingOrientationNotifications;
 #endif
@@ -80,7 +80,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
       return nil;
     }
     NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
     _orientation = UIDeviceOrientationPortrait;
     _rotation = RTCVideoRotation_90;
     [center addObserver:self
@@ -171,7 +171,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
                                  format,
                                  (long)fps);
 
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
                       dispatch_async(dispatch_get_main_queue(), ^{
                         if (!self->_generatingOrientationNotifications) {
                           [[UIDevice currentDevice]
@@ -221,7 +221,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
                       }
                       [self.captureSession stopRunning];
 
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
                       dispatch_async(dispatch_get_main_queue(), ^{
                         if (self->_generatingOrientationNotifications) {
                           [[UIDevice currentDevice]
@@ -239,7 +239,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
 
 #pragma mark iOS notifications
 
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
 - (void)deviceOrientationDidChange:(NSNotification *)notification {
   [RTC_OBJC_TYPE(RTCDispatcher)
       dispatchAsyncOnType:RTCDispatcherTypeCaptureSession
@@ -267,7 +267,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
     return;
   }
 
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
   // Default to portrait orientation on iPhone.
   BOOL usingFrontCamera = NO;
   // Check the image's EXIF for the camera the image came from as the image
@@ -323,7 +323,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
 - (void)captureOutput:(AVCaptureOutput *)captureOutput
     didDropSampleBuffer:(CMSampleBufferRef)sampleBuffer
          fromConnection:(AVCaptureConnection *)connection {
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
   CFStringRef droppedReason = CMGetAttachment(
       sampleBuffer, kCMSampleBufferAttachmentKey_DroppedFrameReason, nil);
 #else
@@ -338,7 +338,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
 
 - (void)handleCaptureSessionInterruption:(NSNotification *)notification {
   NSString *reasonString = nil;
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
   NSNumber *reason =
       notification.userInfo[AVCaptureSessionInterruptionReasonKey];
   if (reason) {
@@ -373,7 +373,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
   [RTC_OBJC_TYPE(RTCDispatcher)
       dispatchAsyncOnType:RTCDispatcherTypeCaptureSession
                     block:^{
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
                       if (error.code == AVErrorMediaServicesWereReset) {
                         [self handleNonFatalError];
                       } else {
@@ -427,7 +427,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
                     }];
 }
 
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
 
 #pragma mark - UIApplication notifications
 
@@ -568,7 +568,7 @@ const int64_t kNanosecondsPerSecond = 1000000000;
   NSAssert([RTC_OBJC_TYPE(RTCDispatcher)
                isOnQueueForType:RTCDispatcherTypeCaptureSession],
            @"updateOrientation must be called on the capture queue.");
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE && !TARGET_OS_TV
   _orientation = [UIDevice currentDevice].orientation;
 #endif
 }
diff --git a/sdk/objc/components/renderer/metal/RTCMTLRenderer.h b/sdk/objc/components/renderer/metal/RTCMTLRenderer.h
index f513fccba1338623c0aaf46826dbce14f8dd822d..ff33cb91d8b0bbfd6b033415ddfcb1cd2ad5f537 100644
--- a/sdk/objc/components/renderer/metal/RTCMTLRenderer.h
+++ b/sdk/objc/components/renderer/metal/RTCMTLRenderer.h
@@ -39,7 +39,7 @@ NS_ASSUME_NONNULL_BEGIN
  * for performing cleanups.
  */
 
-#if TARGET_OS_IOS
+#if TARGET_OS_IOS || TARGET_OS_TV
 - (BOOL)addRenderingDestination:(__kindof UIView *)view;
 #else
 - (BOOL)addRenderingDestination:(__kindof NSView *)view;
diff --git a/sdk/objc/helpers/RTCCameraPreviewView.h b/sdk/objc/helpers/RTCCameraPreviewView.h
index 788e8c3e3eb4be100c5552c500178340d316dd46..3db184fdbb4e6738ebbbb84f61cf4386aab56536 100644
--- a/sdk/objc/helpers/RTCCameraPreviewView.h
+++ b/sdk/objc/helpers/RTCCameraPreviewView.h
@@ -18,6 +18,7 @@
 /** RTCCameraPreviewView is a view that renders local video from an
  *  AVCaptureSession.
  */
+API_AVAILABLE(tvos(17.0), ios(12.0), macos(10.11))
 RTC_OBJC_EXPORT
 @interface RTC_OBJC_TYPE (RTCCameraPreviewView) : UIView
 
diff --git a/sdk/objc/helpers/RTCCameraPreviewView.m b/sdk/objc/helpers/RTCCameraPreviewView.m
index 3314322b992db1e8cfa217de1ae8d8c794df4f5b..d56ca5fd63879d6e7c3793fee0b5b37309d18e67 100644
--- a/sdk/objc/helpers/RTCCameraPreviewView.m
+++ b/sdk/objc/helpers/RTCCameraPreviewView.m
@@ -80,10 +80,11 @@
 }
 
 - (void)setCorrectVideoOrientation {
-  if (@available(iOS 17, *)) {
+  if (@available(iOS 17, tvOS 17, *)) {
     [self modifyVideoAngle];
   }
-#if !defined(__IPHONE_17_0) || __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_17_0
+#if !TARGET_OS_TV && \
+    (!defined(__IPHONE_17_0) || __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_17_0)
   else {
     [self modifyVideoOrientation];
   }
@@ -93,25 +94,29 @@
 #pragma mark - Private
 
 - (void)addOrientationObserver {
+#if !TARGET_OS_TV
   [[NSNotificationCenter defaultCenter]
       addObserver:self
          selector:@selector(orientationChanged:)
              name:UIDeviceOrientationDidChangeNotification
            object:nil];
+#endif
 }
 
 - (void)removeOrientationObserver {
+#if !TARGET_OS_TV
   [[NSNotificationCenter defaultCenter]
       removeObserver:self
                 name:UIDeviceOrientationDidChangeNotification
               object:nil];
+#endif
 }
 
 - (AVCaptureVideoPreviewLayer *)previewLayer {
   return (AVCaptureVideoPreviewLayer *)self.layer;
 }
 
-- (void)modifyVideoAngle API_AVAILABLE(ios(17.0)) {
+- (void)modifyVideoAngle API_AVAILABLE(ios(17.0), tvos(17.0)) {
   AVCaptureDeviceInput* captureSessionInput =
       _captureSession.inputs.firstObject;
   AVCaptureDevice* camera = captureSessionInput.device;
@@ -128,7 +133,8 @@
   }
 }
 
-#if !defined(__IPHONE_17_0) || __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_17_0
+#if !TARGET_OS_TV && \
+    (!defined(__IPHONE_17_0) || __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_17_0)
 - (void)modifyVideoOrientation {
   // Get current device orientation.
   UIDeviceOrientation deviceOrientation = [UIDevice currentDevice].orientation;
@@ -156,4 +162,4 @@
 }
 #endif
 
-@end
\ No newline at end of file
+@end
