diff --git a/sdk/objc/api/peerconnection/RTCAudioTrack.mm b/sdk/objc/api/peerconnection/RTCAudioTrack.mm
index 5ba53c84e9a6b9fb790cffed101d200464df5136..2166b8cb348525a62b2907a2cb0f35c910387faa 100644
--- a/sdk/objc/api/peerconnection/RTCAudioTrack.mm
+++ b/sdk/objc/api/peerconnection/RTCAudioTrack.mm
@@ -29,7 +29,7 @@
   RTC_DCHECK(source);
   RTC_DCHECK(trackId.length);
 
-  std::string nativeId = [NSString stdStringForString:trackId];
+  std::string nativeId = [NSString rtcStdStringForString:trackId];
   webrtc::scoped_refptr<webrtc::AudioTrackInterface> track =
       factory.nativeFactory->CreateAudioTrack(nativeId,
                                               source.nativeAudioSource.get());
diff --git a/sdk/objc/api/peerconnection/RTCDataChannel.mm b/sdk/objc/api/peerconnection/RTCDataChannel.mm
index b52089a854fad9b5004765f96b34586a960c9976..bf86444ff0cf89173b009eaf21950fd125ad1f49 100644
--- a/sdk/objc/api/peerconnection/RTCDataChannel.mm
+++ b/sdk/objc/api/peerconnection/RTCDataChannel.mm
@@ -102,7 +102,7 @@ class DataChannelDelegateAdapter : public DataChannelObserver {
 }
 
 - (NSString *)label {
-  return [NSString stringForStdString:_nativeDataChannel->label()];
+  return [NSString rtcStringForStdString:_nativeDataChannel->label()];
 }
 
 - (BOOL)isReliable {
@@ -134,7 +134,7 @@ class DataChannelDelegateAdapter : public DataChannelObserver {
 }
 
 - (NSString *)protocol {
-  return [NSString stringForStdString:_nativeDataChannel->protocol()];
+  return [NSString rtcStringForStdString:_nativeDataChannel->protocol()];
 }
 
 - (BOOL)isNegotiated {
diff --git a/sdk/objc/api/peerconnection/RTCDataChannelConfiguration.mm b/sdk/objc/api/peerconnection/RTCDataChannelConfiguration.mm
index bf775b1afdd574f2b4cf19649518e0c0493f0dfa..57bd96a75006d39f07c35febabe929f021088e18 100644
--- a/sdk/objc/api/peerconnection/RTCDataChannelConfiguration.mm
+++ b/sdk/objc/api/peerconnection/RTCDataChannelConfiguration.mm
@@ -53,11 +53,11 @@
 }
 
 - (NSString *)protocol {
-  return [NSString stringForStdString:_nativeDataChannelInit.protocol];
+  return [NSString rtcStringForStdString:_nativeDataChannelInit.protocol];
 }
 
 - (void)setProtocol:(NSString *)protocol {
-  _nativeDataChannelInit.protocol = [NSString stdStringForString:protocol];
+  _nativeDataChannelInit.protocol = [NSString rtcStdStringForString:protocol];
 }
 
 - (BOOL)isNegotiated {
diff --git a/sdk/objc/api/peerconnection/RTCDtmfSender.mm b/sdk/objc/api/peerconnection/RTCDtmfSender.mm
index 56b43efbabb98468f49d53ec6818018890b4fbbc..927a6b1b3f74dee0aa0925f82fbaf2fc00ddd396 100644
--- a/sdk/objc/api/peerconnection/RTCDtmfSender.mm
+++ b/sdk/objc/api/peerconnection/RTCDtmfSender.mm
@@ -34,11 +34,11 @@
   int interToneGapMs =
       static_cast<int>(interToneGap * webrtc::kNumMillisecsPerSec);
   return _nativeDtmfSender->InsertDtmf(
-      [NSString stdStringForString:tones], durationMs, interToneGapMs);
+      [NSString rtcStdStringForString:tones], durationMs, interToneGapMs);
 }
 
 - (nonnull NSString *)remainingTones {
-  return [NSString stringForStdString:_nativeDtmfSender->tones()];
+  return [NSString rtcStringForStdString:_nativeDtmfSender->tones()];
 }
 
 - (NSTimeInterval)duration {
diff --git a/sdk/objc/api/peerconnection/RTCIceCandidate.mm b/sdk/objc/api/peerconnection/RTCIceCandidate.mm
index 025601b29561fa72e6ba6932070a3add5845c5ac..cff141fe6e3db2478254e56b63288bc6006cc1c5 100644
--- a/sdk/objc/api/peerconnection/RTCIceCandidate.mm
+++ b/sdk/objc/api/peerconnection/RTCIceCandidate.mm
@@ -52,11 +52,11 @@
   std::string sdp = candidate->ToString();
 
   RTC_OBJC_TYPE(RTCIceCandidate) *rtcCandidate =
-      [self initWithSdp:[NSString stringForStdString:sdp]
+      [self initWithSdp:[NSString rtcStringForStdString:sdp]
           sdpMLineIndex:candidate->sdp_mline_index()
-                 sdpMid:[NSString stringForStdString:candidate->sdp_mid()]];
+                 sdpMid:[NSString rtcStringForStdString:candidate->sdp_mid()]];
   rtcCandidate->_serverUrl =
-      [NSString stringForStdString:candidate->server_url()];
+      [NSString rtcStringForStdString:candidate->server_url()];
   return rtcCandidate;
 }
 
@@ -64,7 +64,7 @@
   webrtc::SdpParseError error;
 
   webrtc::IceCandidate *candidate = webrtc::CreateIceCandidate(
-      _sdpMid.stdString, _sdpMLineIndex, _sdp.stdString, &error);
+      _sdpMid.rtcStdString, _sdpMLineIndex, _sdp.rtcStdString, &error);
 
   if (!candidate) {
     RTCLog(@"Failed to create ICE candidate: %s\nline: %s",
diff --git a/sdk/objc/api/peerconnection/RTCIceCandidateErrorEvent.mm b/sdk/objc/api/peerconnection/RTCIceCandidateErrorEvent.mm
index cef03f2f438ac37c6e66b9d2877312b1ece6c1ce..0b0b01dd5db39c15e96f60b1aa696b3984c91ed8 100644
--- a/sdk/objc/api/peerconnection/RTCIceCandidateErrorEvent.mm
+++ b/sdk/objc/api/peerconnection/RTCIceCandidateErrorEvent.mm
@@ -31,11 +31,11 @@
                       errorText:(const std::string&)errorText {
   self = [self init];
   if (self) {
-    _address = [NSString stringForStdString:address];
+    _address = [NSString rtcStringForStdString:address];
     _port = port;
-    _url = [NSString stringForStdString:url];
+    _url = [NSString rtcStringForStdString:url];
     _errorCode = errorCode;
-    _errorText = [NSString stringForStdString:errorText];
+    _errorText = [NSString rtcStringForStdString:errorText];
   }
   return self;
 }
diff --git a/sdk/objc/api/peerconnection/RTCIceServer.mm b/sdk/objc/api/peerconnection/RTCIceServer.mm
index 96961e33d0bf4351aeb83cc6a4a1874343dd1c99..34e81ec0993472d94c187c86d5e92201acd1748f 100644
--- a/sdk/objc/api/peerconnection/RTCIceServer.mm
+++ b/sdk/objc/api/peerconnection/RTCIceServer.mm
@@ -124,23 +124,23 @@
 - (webrtc::PeerConnectionInterface::IceServer)nativeServer {
   __block webrtc::PeerConnectionInterface::IceServer iceServer;
 
-  iceServer.username = [NSString stdStringForString:_username];
-  iceServer.password = [NSString stdStringForString:_credential];
-  iceServer.hostname = [NSString stdStringForString:_hostname];
+  iceServer.username = [NSString rtcStdStringForString:_username];
+  iceServer.password = [NSString rtcStdStringForString:_credential];
+  iceServer.hostname = [NSString rtcStdStringForString:_hostname];
 
   [_tlsAlpnProtocols enumerateObjectsUsingBlock:^(
                          NSString *proto, NSUInteger idx, BOOL *stop) {
-    iceServer.tls_alpn_protocols.push_back(proto.stdString);
+    iceServer.tls_alpn_protocols.push_back(proto.rtcStdString);
   }];
 
   [_tlsEllipticCurves enumerateObjectsUsingBlock:^(
                           NSString *curve, NSUInteger idx, BOOL *stop) {
-    iceServer.tls_elliptic_curves.push_back(curve.stdString);
+    iceServer.tls_elliptic_curves.push_back(curve.rtcStdString);
   }];
 
   [_urlStrings
       enumerateObjectsUsingBlock:^(NSString *url, NSUInteger idx, BOOL *stop) {
-        iceServer.urls.push_back(url.stdString);
+        iceServer.urls.push_back(url.rtcStdString);
       }];
 
   switch (_tlsCertPolicy) {
@@ -161,20 +161,20 @@
   NSMutableArray *urls =
       [NSMutableArray arrayWithCapacity:nativeServer.urls.size()];
   for (auto const &url : nativeServer.urls) {
-    [urls addObject:[NSString stringForStdString:url]];
+    [urls addObject:[NSString rtcStringForStdString:url]];
   }
-  NSString *username = [NSString stringForStdString:nativeServer.username];
-  NSString *credential = [NSString stringForStdString:nativeServer.password];
-  NSString *hostname = [NSString stringForStdString:nativeServer.hostname];
+  NSString *username = [NSString rtcStringForStdString:nativeServer.username];
+  NSString *credential = [NSString rtcStringForStdString:nativeServer.password];
+  NSString *hostname = [NSString rtcStringForStdString:nativeServer.hostname];
   NSMutableArray *tlsAlpnProtocols =
       [NSMutableArray arrayWithCapacity:nativeServer.tls_alpn_protocols.size()];
   for (auto const &proto : nativeServer.tls_alpn_protocols) {
-    [tlsAlpnProtocols addObject:[NSString stringForStdString:proto]];
+    [tlsAlpnProtocols addObject:[NSString rtcStringForStdString:proto]];
   }
   NSMutableArray *tlsEllipticCurves = [NSMutableArray
       arrayWithCapacity:nativeServer.tls_elliptic_curves.size()];
   for (auto const &curve : nativeServer.tls_elliptic_curves) {
-    [tlsEllipticCurves addObject:[NSString stringForStdString:curve]];
+    [tlsEllipticCurves addObject:[NSString rtcStringForStdString:curve]];
   }
   RTCTlsCertPolicy tlsCertPolicy;
 
diff --git a/sdk/objc/api/peerconnection/RTCLegacyStatsReport.mm b/sdk/objc/api/peerconnection/RTCLegacyStatsReport.mm
index 703b44c041f2fd6b57adc9331a55ffdf780ed65a..1f34ef8dc9a72d71420b2130122fdda3b21f7a3e 100644
--- a/sdk/objc/api/peerconnection/RTCLegacyStatsReport.mm
+++ b/sdk/objc/api/peerconnection/RTCLegacyStatsReport.mm
@@ -37,17 +37,17 @@
   self = [super init];
   if (self) {
     _timestamp = nativeReport.timestamp();
-    _type = [NSString stringForStdString:nativeReport.TypeToString()];
-    _reportId = [NSString stringForStdString:nativeReport.id()->ToString()];
+    _type = [NSString rtcStringForStdString:nativeReport.TypeToString()];
+    _reportId = [NSString rtcStringForStdString:nativeReport.id()->ToString()];
 
     NSUInteger capacity = nativeReport.values().size();
     NSMutableDictionary *values =
         [NSMutableDictionary dictionaryWithCapacity:capacity];
     for (auto const &valuePair : nativeReport.values()) {
       NSString *key =
-          [NSString stringForStdString:valuePair.second->display_name()];
+          [NSString rtcStringForStdString:valuePair.second->display_name()];
       NSString *value =
-          [NSString stringForStdString:valuePair.second->ToString()];
+          [NSString rtcStringForStdString:valuePair.second->ToString()];
 
       // Not expecting duplicate keys.
       RTC_DCHECK(![values objectForKey:key]);
diff --git a/sdk/objc/api/peerconnection/RTCMediaConstraints.mm b/sdk/objc/api/peerconnection/RTCMediaConstraints.mm
index 5158b5905bbca5be08ea01961a60015c267fe0a0..28445ed8d31dacec324e5213b453d5c07b3c71e1 100644
--- a/sdk/objc/api/peerconnection/RTCMediaConstraints.mm
+++ b/sdk/objc/api/peerconnection/RTCMediaConstraints.mm
@@ -87,10 +87,10 @@ NSString *const kRTCMediaConstraintsValueFalse =
       std::string configValue = std::string(
           reinterpret_cast<const char *>(charData.bytes), charData.length);
       nativeConstraints.push_back(
-          webrtc::MediaConstraints::Constraint(key.stdString, configValue));
+          webrtc::MediaConstraints::Constraint(key.rtcStdString, configValue));
     } else {
       nativeConstraints.push_back(
-          webrtc::MediaConstraints::Constraint(key.stdString, value.stdString));
+          webrtc::MediaConstraints::Constraint(key.rtcStdString, value.rtcStdString));
     }
   }
   return nativeConstraints;
diff --git a/sdk/objc/api/peerconnection/RTCMediaStream.mm b/sdk/objc/api/peerconnection/RTCMediaStream.mm
index 60c256cc94e4d8adc7bb4575371c6894bc8f3979..9bcc3f02c5bb16946b141aa8d4e5c327a3787307 100644
--- a/sdk/objc/api/peerconnection/RTCMediaStream.mm
+++ b/sdk/objc/api/peerconnection/RTCMediaStream.mm
@@ -29,7 +29,7 @@
                        streamId:(NSString *)streamId {
   NSParameterAssert(factory);
   NSParameterAssert(streamId.length);
-  std::string nativeId = [NSString stdStringForString:streamId];
+  std::string nativeId = [NSString rtcStdStringForString:streamId];
   webrtc::scoped_refptr<webrtc::MediaStreamInterface> stream =
       factory.nativeFactory->CreateLocalMediaStream(nativeId);
   return [self initWithFactory:factory nativeMediaStream:stream];
@@ -52,7 +52,7 @@
 }
 
 - (NSString *)streamId {
-  return [NSString stringForStdString:_nativeMediaStream->id()];
+  return [NSString rtcStringForStdString:_nativeMediaStream->id()];
 }
 
 - (void)addAudioTrack:(RTC_OBJC_TYPE(RTCAudioTrack) *)audioTrack {
diff --git a/sdk/objc/api/peerconnection/RTCMediaStreamTrack.mm b/sdk/objc/api/peerconnection/RTCMediaStreamTrack.mm
index 244b5e7207214457d252edf0862c7d248dde564f..7648eede0d5f191b7206c05c953dde67ecb50ffd 100644
--- a/sdk/objc/api/peerconnection/RTCMediaStreamTrack.mm
+++ b/sdk/objc/api/peerconnection/RTCMediaStreamTrack.mm
@@ -26,11 +26,11 @@ NSString *const kRTCMediaStreamTrackKindVideo =
 }
 
 - (NSString *)kind {
-  return [NSString stringForStdString:_nativeTrack->kind()];
+  return [NSString rtcStringForStdString:_nativeTrack->kind()];
 }
 
 - (NSString *)trackId {
-  return [NSString stringForStdString:_nativeTrack->id()];
+  return [NSString rtcStringForStdString:_nativeTrack->id()];
 }
 
 - (BOOL)isEnabled {
diff --git a/sdk/objc/api/peerconnection/RTCMetricsSampleInfo.mm b/sdk/objc/api/peerconnection/RTCMetricsSampleInfo.mm
index ea01350122a86843254b3117a03863ba131ee77b..4c7597ded603cfe148256122ae6ee2ed6d2cdfab 100644
--- a/sdk/objc/api/peerconnection/RTCMetricsSampleInfo.mm
+++ b/sdk/objc/api/peerconnection/RTCMetricsSampleInfo.mm
@@ -26,7 +26,7 @@
     (const webrtc::metrics::SampleInfo &)info {
   self = [super init];
   if (self) {
-    _name = [NSString stringForStdString:info.name];
+    _name = [NSString rtcStringForStdString:info.name];
     _min = info.min;
     _max = info.max;
     _bucketCount = info.bucket_count;
diff --git a/sdk/objc/api/peerconnection/RTCPeerConnection+DataChannel.mm b/sdk/objc/api/peerconnection/RTCPeerConnection+DataChannel.mm
index ea5659877428f35c0286e6c2bce1aa621d7c65c4..02ea9235c3e88714e8cfab9d2c23900301c25fa3 100644
--- a/sdk/objc/api/peerconnection/RTCPeerConnection+DataChannel.mm
+++ b/sdk/objc/api/peerconnection/RTCPeerConnection+DataChannel.mm
@@ -20,7 +20,7 @@
     - (nullable RTC_OBJC_TYPE(RTCDataChannel) *)dataChannelForLabel
     : (NSString *)label configuration
     : (RTC_OBJC_TYPE(RTCDataChannelConfiguration) *)configuration {
-  std::string labelString = [NSString stdStringForString:label];
+  std::string labelString = [NSString rtcStdStringForString:label];
   const webrtc::DataChannelInit nativeInit =
       configuration.nativeDataChannelInit;
   auto result = self.nativePeerConnection->CreateDataChannelOrError(
diff --git a/sdk/objc/api/peerconnection/RTCPeerConnection.mm b/sdk/objc/api/peerconnection/RTCPeerConnection.mm
index 093e95706e83d698e6c467c69550ac3c76fd71dc..3319424cab20e767f569fe94f5898b9cafb11914 100644
--- a/sdk/objc/api/peerconnection/RTCPeerConnection.mm
+++ b/sdk/objc/api/peerconnection/RTCPeerConnection.mm
@@ -67,7 +67,7 @@ class SetSessionDescriptionObserver
       completion_handler_(nil);
     } else {
       // TODO(hta): Add handling of error.type()
-      NSString *str = [NSString stringForStdString:error.message()];
+      NSString *str = [NSString rtcStringForStdString:error.message()];
       NSError *err =
           [NSError errorWithDomain:kRTCPeerConnectionErrorDomain
                               code:kRTCPeerConnnectionSessionDescriptionError
@@ -110,7 +110,7 @@ class CreateSessionDescriptionObserverAdapter
   void OnFailure(RTCError error) override {
     RTC_DCHECK(completion_handler_);
     // TODO(hta): Add handling of error.type()
-    NSString *str = [NSString stringForStdString:error.message()];
+    NSString *str = [NSString rtcStringForStdString:error.message()];
     NSError *err =
         [NSError errorWithDomain:kRTCPeerConnectionErrorDomain
                             code:kRTCPeerConnnectionSessionDescriptionError
@@ -400,7 +400,7 @@ void PeerConnectionDelegateAdapter::OnIceSelectedCandidatePairChanged(
   RTC_OBJC_TYPE(RTCIceCandidate) *remote_candidate =
       [[RTC_OBJC_TYPE(RTCIceCandidate) alloc]
           initWithNativeCandidate:&remote_candidate_wrapper];
-  NSString *nsstr_reason = [NSString stringForStdString:event.reason];
+  NSString *nsstr_reason = [NSString rtcStringForStdString:event.reason];
   if ([delegate respondsToSelector:@selector
                 (peerConnection:
                     didChangeLocalCandidate:remoteCandidate:lastReceivedMs
@@ -621,7 +621,7 @@ void PeerConnectionDelegateAdapter::OnRemoveTrack(
         if (error.ok()) {
           completionHandler(nil);
         } else {
-          NSString *str = [NSString stringForStdString:error.message()];
+          NSString *str = [NSString rtcStringForStdString:error.message()];
           NSError *err =
               [NSError errorWithDomain:kRTCPeerConnectionErrorDomain
                                   code:static_cast<NSInteger>(error.type())
@@ -845,8 +845,8 @@ void PeerConnectionDelegateAdapter::OnRemoveTrack(
 
 - (RTC_OBJC_TYPE(RTCRtpSender) *)senderWithKind:(NSString *)kind
                                        streamId:(NSString *)streamId {
-  std::string nativeKind = [NSString stdStringForString:kind];
-  std::string nativeStreamId = [NSString stdStringForString:streamId];
+  std::string nativeKind = [NSString rtcStdStringForString:kind];
+  std::string nativeStreamId = [NSString rtcStdStringForString:streamId];
   RTC_ALLOW_PLAN_B_DEPRECATION_BEGIN();
   webrtc::scoped_refptr<webrtc::RtpSenderInterface> nativeSender(
       _peerConnection->CreateSender(nativeKind, nativeStreamId));
diff --git a/sdk/objc/api/peerconnection/RTCRtcpParameters.mm b/sdk/objc/api/peerconnection/RTCRtcpParameters.mm
index 7bb5d855945d6943e27d89bee8f16d2377b56e01..9b6f22cd9cde0d7800f3456f7860a9500accfe08 100644
--- a/sdk/objc/api/peerconnection/RTCRtcpParameters.mm
+++ b/sdk/objc/api/peerconnection/RTCRtcpParameters.mm
@@ -26,7 +26,7 @@
     (const webrtc::RtcpParameters &)nativeParameters {
   self = [super init];
   if (self) {
-    _cname = [NSString stringForStdString:nativeParameters.cname];
+    _cname = [NSString rtcStringForStdString:nativeParameters.cname];
     _isReducedSize = nativeParameters.reduced_size;
   }
   return self;
@@ -34,7 +34,7 @@
 
 - (webrtc::RtcpParameters)nativeParameters {
   webrtc::RtcpParameters parameters;
-  parameters.cname = [NSString stdStringForString:_cname];
+  parameters.cname = [NSString rtcStdStringForString:_cname];
   parameters.reduced_size = _isReducedSize;
   return parameters;
 }
diff --git a/sdk/objc/api/peerconnection/RTCRtpCodecCapability.mm b/sdk/objc/api/peerconnection/RTCRtpCodecCapability.mm
index 0cfe53a67720730e53344447c515e761939655cc..7fe4cef6b1d257287c177810db0d7f5779b9c4ae 100644
--- a/sdk/objc/api/peerconnection/RTCRtpCodecCapability.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpCodecCapability.mm
@@ -39,7 +39,7 @@
       _preferredPayloadType = [NSNumber
           numberWithInt:*nativeRtpCodecCapability.preferred_payload_type];
     }
-    _name = [NSString stringForStdString:nativeRtpCodecCapability.name];
+    _name = [NSString rtcStringForStdString:nativeRtpCodecCapability.name];
     switch (nativeRtpCodecCapability.kind) {
       case webrtc::MediaType::AUDIO:
         _kind = kRTCMediaStreamTrackKindAudio;
@@ -61,12 +61,12 @@
     }
     NSMutableDictionary *parameters = [NSMutableDictionary dictionary];
     for (const auto &parameter : nativeRtpCodecCapability.parameters) {
-      [parameters setObject:[NSString stringForStdString:parameter.second]
-                     forKey:[NSString stringForStdString:parameter.first]];
+      [parameters setObject:[NSString rtcStringForStdString:parameter.second]
+                     forKey:[NSString rtcStringForStdString:parameter.first]];
     }
     _parameters = parameters;
     _mimeType =
-        [NSString stringForStdString:nativeRtpCodecCapability.mime_type()];
+        [NSString rtcStringForStdString:nativeRtpCodecCapability.mime_type()];
   }
   return self;
 }
@@ -92,7 +92,7 @@
     rtpCodecCapability.preferred_payload_type =
         std::optional<int>(_preferredPayloadType.intValue);
   }
-  rtpCodecCapability.name = [NSString stdStringForString:_name];
+  rtpCodecCapability.name = [NSString rtcStdStringForString:_name];
   // NSString pointer comparison is safe here since "kind" is readonly and only
   // populated above.
   if (_kind == kRTCMediaStreamTrackKindAudio) {
@@ -109,8 +109,8 @@
     rtpCodecCapability.num_channels = std::optional<int>(_numChannels.intValue);
   }
   for (NSString *paramKey in _parameters.allKeys) {
-    std::string key = [NSString stdStringForString:paramKey];
-    std::string value = [NSString stdStringForString:_parameters[paramKey]];
+    std::string key = [NSString rtcStdStringForString:paramKey];
+    std::string value = [NSString rtcStdStringForString:_parameters[paramKey]];
     rtpCodecCapability.parameters[key] = value;
   }
   return rtpCodecCapability;
diff --git a/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm b/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
index deb4738f5bd7264683cac576bcdcf2cb1d2a0f22..aa74bb1f2e15e7286c9c8ae2309026ead426ee1d 100644
--- a/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
@@ -51,7 +51,7 @@ const NSString *const kRTCH264CodecName = @(webrtc::kH264CodecName);
   self = [super init];
   if (self) {
     _payloadType = nativeParameters.payload_type;
-    _name = [NSString stringForStdString:nativeParameters.name];
+    _name = [NSString rtcStringForStdString:nativeParameters.name];
     switch (nativeParameters.kind) {
       case webrtc::MediaType::AUDIO:
         _kind = kRTCMediaStreamTrackKindAudio;
@@ -71,8 +71,8 @@ const NSString *const kRTCH264CodecName = @(webrtc::kH264CodecName);
     }
     NSMutableDictionary *parameters = [NSMutableDictionary dictionary];
     for (const auto &parameter : nativeParameters.parameters) {
-      [parameters setObject:[NSString stringForStdString:parameter.second]
-                     forKey:[NSString stringForStdString:parameter.first]];
+      [parameters setObject:[NSString rtcStringForStdString:parameter.second]
+                     forKey:[NSString rtcStringForStdString:parameter.first]];
     }
     _parameters = parameters;
   }
@@ -82,7 +82,7 @@ const NSString *const kRTCH264CodecName = @(webrtc::kH264CodecName);
 - (webrtc::RtpCodecParameters)nativeParameters {
   webrtc::RtpCodecParameters parameters;
   parameters.payload_type = _payloadType;
-  parameters.name = [NSString stdStringForString:_name];
+  parameters.name = [NSString rtcStdStringForString:_name];
   // NSString pointer comparison is safe here since "kind" is readonly and only
   // populated above.
   if (_kind == kRTCMediaStreamTrackKindAudio) {
@@ -99,8 +99,8 @@ const NSString *const kRTCH264CodecName = @(webrtc::kH264CodecName);
     parameters.num_channels = std::optional<int>(_numChannels.intValue);
   }
   for (NSString *paramKey in _parameters.allKeys) {
-    std::string key = [NSString stdStringForString:paramKey];
-    std::string value = [NSString stdStringForString:_parameters[paramKey]];
+    std::string key = [NSString rtcStdStringForString:paramKey];
+    std::string value = [NSString rtcStdStringForString:_parameters[paramKey]];
     parameters.parameters[key] = value;
   }
   return parameters;
diff --git a/sdk/objc/api/peerconnection/RTCRtpEncodingParameters.mm b/sdk/objc/api/peerconnection/RTCRtpEncodingParameters.mm
index 6fda1a516025f52a01514ece345363313a00cb94..a122136e1fd07b2ed9d1e7ffed9e311582818f93 100644
--- a/sdk/objc/api/peerconnection/RTCRtpEncodingParameters.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpEncodingParameters.mm
@@ -37,7 +37,7 @@
   self = [super init];
   if (self) {
     if (!nativeParameters.rid.empty()) {
-      _rid = [NSString stringForStdString:nativeParameters.rid];
+      _rid = [NSString rtcStringForStdString:nativeParameters.rid];
     }
     _isActive = nativeParameters.active;
     if (nativeParameters.max_bitrate_bps) {
@@ -57,7 +57,7 @@
     }
     if (nativeParameters.scalability_mode) {
       _scalabilityMode =
-          [NSString stringForStdString:*nativeParameters.scalability_mode];
+          [NSString rtcStringForStdString:*nativeParameters.scalability_mode];
     }
     if (nativeParameters.scale_resolution_down_by) {
       _scaleResolutionDownBy = [NSNumber
@@ -77,7 +77,7 @@
 - (webrtc::RtpEncodingParameters)nativeParameters {
   webrtc::RtpEncodingParameters parameters;
   if (_rid != nil) {
-    parameters.rid = [NSString stdStringForString:_rid];
+    parameters.rid = [NSString rtcStdStringForString:_rid];
   }
   parameters.active = _isActive;
   if (_maxBitrateBps != nil) {
@@ -95,7 +95,7 @@
   }
   if (_scalabilityMode != nil) {
     parameters.scalability_mode = std::optional<std::string>(
-        [NSString stdStringForString:_scalabilityMode]);
+        [NSString rtcStdStringForString:_scalabilityMode]);
   }
   if (_scaleResolutionDownBy != nil) {
     parameters.scale_resolution_down_by =
diff --git a/sdk/objc/api/peerconnection/RTCRtpHeaderExtension.mm b/sdk/objc/api/peerconnection/RTCRtpHeaderExtension.mm
index 88f26506f3e1b4433af5c138c4759c3f331bfd19..664ba24940ee1c1769a882a27bf89a69b0c514ba 100644
--- a/sdk/objc/api/peerconnection/RTCRtpHeaderExtension.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpHeaderExtension.mm
@@ -27,7 +27,7 @@
     (const webrtc::RtpExtension &)nativeParameters {
   self = [super init];
   if (self) {
-    _uri = [NSString stringForStdString:nativeParameters.uri];
+    _uri = [NSString rtcStringForStdString:nativeParameters.uri];
     _id = nativeParameters.id.value();
     _encrypted = nativeParameters.encrypt;
   }
@@ -36,7 +36,7 @@
 
 - (webrtc::RtpExtension)nativeParameters {
   webrtc::RtpExtension extension;
-  extension.uri = [NSString stdStringForString:_uri];
+  extension.uri = [NSString rtcStdStringForString:_uri];
   extension.id = webrtc::RtpHeaderExtensionId(_id);
   extension.encrypt = _encrypted;
   return extension;
diff --git a/sdk/objc/api/peerconnection/RTCRtpHeaderExtensionCapability.mm b/sdk/objc/api/peerconnection/RTCRtpHeaderExtensionCapability.mm
index ef40aca4df0748f9e408cf880d3f3321bae8c7b1..bdbcd6c90b0fa327a32155f60a6510249dd0be37 100644
--- a/sdk/objc/api/peerconnection/RTCRtpHeaderExtensionCapability.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpHeaderExtensionCapability.mm
@@ -31,7 +31,7 @@
         nativeRtpHeaderExtensionCapability {
   self = [super init];
   if (self) {
-    _uri = [NSString stringForStdString:nativeRtpHeaderExtensionCapability.uri];
+    _uri = [NSString rtcStringForStdString:nativeRtpHeaderExtensionCapability.uri];
     if (nativeRtpHeaderExtensionCapability.preferred_id) {
       _preferredId = [NSNumber numberWithInt:nativeRtpHeaderExtensionCapability
                                                  .preferred_id->value()];
@@ -59,13 +59,13 @@
       nativeRtpTransceiverDirectionFromDirection:_direction];
   if (_preferredId != nil) {
     return webrtc::RtpHeaderExtensionCapability(
-        [NSString stdStringForString:_uri],
+        [NSString rtcStdStringForString:_uri],
         webrtc::RtpHeaderExtensionId(_preferredId.intValue),
         _preferredEncrypted,
         direction);
   } else {
     return webrtc::RtpHeaderExtensionCapability(
-        [NSString stdStringForString:_uri], _preferredEncrypted, direction);
+        [NSString rtcStdStringForString:_uri], _preferredEncrypted, direction);
   }
 }
 
diff --git a/sdk/objc/api/peerconnection/RTCRtpParameters.mm b/sdk/objc/api/peerconnection/RTCRtpParameters.mm
index 2d192abd1c112d9c8431aac0757345dc8ac3538d..c1ff92a088831d36516ecb92e266afe5c83179b6 100644
--- a/sdk/objc/api/peerconnection/RTCRtpParameters.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpParameters.mm
@@ -35,7 +35,7 @@
   self = [super init];
   if (self) {
     _transactionId =
-        [NSString stringForStdString:nativeParameters.transaction_id];
+        [NSString rtcStringForStdString:nativeParameters.transaction_id];
     _rtcp = [[RTC_OBJC_TYPE(RTCRtcpParameters) alloc]
         initWithNativeParameters:nativeParameters.rtcp];
 
@@ -70,7 +70,7 @@
 
 - (webrtc::RtpParameters)nativeParameters {
   webrtc::RtpParameters parameters;
-  parameters.transaction_id = [NSString stdStringForString:_transactionId];
+  parameters.transaction_id = [NSString rtcStdStringForString:_transactionId];
   parameters.rtcp = [_rtcp nativeParameters];
   for (RTC_OBJC_TYPE(RTCRtpHeaderExtension) *
        headerExtension in _headerExtensions) {
diff --git a/sdk/objc/api/peerconnection/RTCRtpReceiver.mm b/sdk/objc/api/peerconnection/RTCRtpReceiver.mm
index 19bae4b2f4f063e6b07622835179ef2917bd8cf1..d0730b6491617d59d8415bc45037140c1f7390a9 100644
--- a/sdk/objc/api/peerconnection/RTCRtpReceiver.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpReceiver.mm
@@ -62,7 +62,7 @@ void RtpReceiverDelegateAdapter::OnFirstPacketReceivedAfterReceptiveChange(
 @synthesize delegate = _delegate;
 
 - (NSString *)receiverId {
-  return [NSString stringForStdString:_nativeRtpReceiver->id()];
+  return [NSString rtcStringForStdString:_nativeRtpReceiver->id()];
 }
 
 - (RTC_OBJC_TYPE(RTCRtpParameters) *)parameters {
diff --git a/sdk/objc/api/peerconnection/RTCRtpSender.mm b/sdk/objc/api/peerconnection/RTCRtpSender.mm
index f3135c6e5ca6d320c3882882ba0866cd0c2ba146..c52f1a69a4f5fe3c44de76d1818d8ea3d4fe60ad 100644
--- a/sdk/objc/api/peerconnection/RTCRtpSender.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpSender.mm
@@ -27,7 +27,7 @@
 @synthesize dtmfSender = _dtmfSender;
 
 - (NSString *)senderId {
-  return [NSString stringForStdString:_nativeRtpSender->id()];
+  return [NSString rtcStringForStdString:_nativeRtpSender->id()];
 }
 
 - (RTC_OBJC_TYPE(RTCRtpParameters) *)parameters {
@@ -68,7 +68,7 @@
   NSMutableArray *streamIds =
       [NSMutableArray arrayWithCapacity:nativeStreamIds.size()];
   for (const auto &s : nativeStreamIds) {
-    [streamIds addObject:[NSString stringForStdString:s]];
+    [streamIds addObject:[NSString rtcStringForStdString:s]];
   }
   return streamIds;
 }
diff --git a/sdk/objc/api/peerconnection/RTCRtpTransceiver.mm b/sdk/objc/api/peerconnection/RTCRtpTransceiver.mm
index faa6b64dd42646b57c4a996b2978af3a5dc08475..c25bf432b3696fc462c8f22ae0cc99a01d9ba2f6 100644
--- a/sdk/objc/api/peerconnection/RTCRtpTransceiver.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpTransceiver.mm
@@ -65,7 +65,7 @@ NSString *const kRTCRtpTransceiverErrorDomain = @"org.webrtc.RTCRtpTranceiver";
 
 - (NSString *)mid {
   if (_nativeRtpTransceiver->mid()) {
-    return [NSString stringForStdString:*_nativeRtpTransceiver->mid()];
+    return [NSString rtcStringForStdString:*_nativeRtpTransceiver->mid()];
   } else {
     return nil;
   }
diff --git a/sdk/objc/api/peerconnection/RTCSessionDescription.mm b/sdk/objc/api/peerconnection/RTCSessionDescription.mm
index b81946fe2a649fe5b378928ffbe36b87d04ad3b4..1c9daf0c60b45a474fc6abb97d4e6156b49a890f 100644
--- a/sdk/objc/api/peerconnection/RTCSessionDescription.mm
+++ b/sdk/objc/api/peerconnection/RTCSessionDescription.mm
@@ -23,11 +23,11 @@
 
 + (NSString *)stringForType:(RTCSdpType)type {
   std::string string = [[self class] stdStringForType:type];
-  return [NSString stringForStdString:string];
+  return [NSString rtcStringForStdString:string];
 }
 
 + (RTCSdpType)typeForString:(NSString *)string {
-  std::string typeString = string.stdString;
+  std::string typeString = string.rtcStdString;
   return [[self class] typeForStdString:typeString];
 }
 
@@ -54,7 +54,7 @@
 
   std::unique_ptr<webrtc::SessionDescriptionInterface> description(
       webrtc::CreateSessionDescription(
-          [[self class] nativeTypeForType:_type], _sdp.stdString, &error));
+          [[self class] nativeTypeForType:_type], _sdp.rtcStdString, &error));
 
   if (!description) {
     RTCLogError(@"Failed to create session description: %s\nline: %s",
@@ -72,7 +72,7 @@
   nativeDescription->ToString(&sdp);
   RTCSdpType type = [[self class] typeForSdpType:nativeDescription->GetType()];
 
-  return [self initWithType:type sdp:[NSString stringForStdString:sdp]];
+  return [self initWithType:type sdp:[NSString rtcStringForStdString:sdp]];
 }
 
 + (std::string)stdStringForType:(RTCSdpType)type {
diff --git a/sdk/objc/api/peerconnection/RTCStatisticsReport.mm b/sdk/objc/api/peerconnection/RTCStatisticsReport.mm
index 68aac57f97678d0079c7535721d9c73bfcb462ef..f2286c6342328bd4afc44554f0144fb6634db336 100644
--- a/sdk/objc/api/peerconnection/RTCStatisticsReport.mm
+++ b/sdk/objc/api/peerconnection/RTCStatisticsReport.mm
@@ -35,7 +35,7 @@ NSObject *ValueFromStatsAttribute(const Attribute &attribute) {
   } else if (attribute.holds_alternative<double>()) {
     return [NSNumber numberWithDouble:attribute.get<double>()];
   } else if (attribute.holds_alternative<std::string>()) {
-    return [NSString stringForStdString:attribute.get<std::string>()];
+    return [NSString rtcStringForStdString:attribute.get<std::string>()];
   } else if (attribute.holds_alternative<std::vector<bool>>()) {
     std::vector<bool> sequence = attribute.get<std::vector<bool>>();
     NSMutableArray *array = [NSMutableArray arrayWithCapacity:sequence.size()];
@@ -89,7 +89,7 @@ NSObject *ValueFromStatsAttribute(const Attribute &attribute) {
     NSMutableArray<NSString *> *array =
         [NSMutableArray arrayWithCapacity:sequence.size()];
     for (const auto &item : sequence) {
-      [array addObject:[NSString stringForStdString:item]];
+      [array addObject:[NSString rtcStringForStdString:item]];
     }
     return [array copy];
   } else if (attribute.holds_alternative<std::map<std::string, uint64_t>>()) {
@@ -98,7 +98,7 @@ NSObject *ValueFromStatsAttribute(const Attribute &attribute) {
     NSMutableDictionary<NSString *, NSNumber *> *dictionary =
         [NSMutableDictionary dictionaryWithCapacity:map.size()];
     for (const auto &item : map) {
-      dictionary[[NSString stringForStdString:item.first]] = @(item.second);
+      dictionary[[NSString rtcStringForStdString:item.first]] = @(item.second);
     }
     return [dictionary copy];
   } else if (attribute.holds_alternative<std::map<std::string, double>>()) {
@@ -107,7 +107,7 @@ NSObject *ValueFromStatsAttribute(const Attribute &attribute) {
     NSMutableDictionary<NSString *, NSNumber *> *dictionary =
         [NSMutableDictionary dictionaryWithCapacity:map.size()];
     for (const auto &item : map) {
-      dictionary[[NSString stringForStdString:item.first]] = @(item.second);
+      dictionary[[NSString rtcStringForStdString:item.first]] = @(item.second);
     }
     return [dictionary copy];
   }
@@ -126,7 +126,7 @@ NSObject *ValueFromStatsAttribute(const Attribute &attribute) {
 - (instancetype)initWithStatistics:(const webrtc::RTCStats &)statistics {
   self = [super init];
   if (self) {
-    _id = [NSString stringForStdString:statistics.id()];
+    _id = [NSString rtcStringForStdString:statistics.id()];
     _timestamp_us = statistics.timestamp().us();
     _type = [NSString stringWithCString:statistics.type()
                                encoding:NSUTF8StringEncoding];
diff --git a/sdk/objc/api/peerconnection/RTCVideoCodecInfo+Private.mm b/sdk/objc/api/peerconnection/RTCVideoCodecInfo+Private.mm
index 8a977a3447909d08900a3349889adde35bcfb64d..78f6d0ca7d2dcc6ec093513e34e46a1ac08aa012 100644
--- a/sdk/objc/api/peerconnection/RTCVideoCodecInfo+Private.mm
+++ b/sdk/objc/api/peerconnection/RTCVideoCodecInfo+Private.mm
@@ -21,8 +21,8 @@
     : (webrtc::SdpVideoFormat)format {
   NSMutableDictionary* params = [NSMutableDictionary dictionary];
   for (const auto& [key, value] : format.parameters) {
-    [params setObject:[NSString stringForStdString:value]
-               forKey:[NSString stringForStdString:key]];
+    [params setObject:[NSString rtcStringForStdString:value]
+               forKey:[NSString rtcStringForStdString:key]];
   }
 
   NSMutableArray<NSString*>* scalabilityModes =
@@ -33,7 +33,7 @@
                       stringForAbslStringView:ScalabilityModeToString(mode)]];
   }
 
-  return [self initWithName:[NSString stringForStdString:format.name]
+  return [self initWithName:[NSString rtcStringForStdString:format.name]
                  parameters:params
            scalabilityModes:scalabilityModes];
 }
@@ -41,8 +41,8 @@
 - (webrtc::SdpVideoFormat)nativeSdpVideoFormat {
   std::map<std::string, std::string> parameters;
   for (NSString* paramKey in self.parameters.allKeys) {
-    std::string key = [NSString stdStringForString:paramKey];
-    std::string value = [NSString stdStringForString:self.parameters[paramKey]];
+    std::string key = [NSString rtcStdStringForString:paramKey];
+    std::string value = [NSString rtcStdStringForString:self.parameters[paramKey]];
     parameters[key] = value;
   }
 
@@ -51,14 +51,14 @@
   for (NSString* mode_name in self.scalabilityModes) {
     std::optional<webrtc::ScalabilityMode> mode =
         webrtc::ScalabilityModeStringToEnum(
-            [NSString stdStringForString:mode_name]);
+            [NSString rtcStdStringForString:mode_name]);
     if (mode.has_value()) {
       scalability_modes.push_back(*mode);
     }
   }
 
   return webrtc::SdpVideoFormat(
-      [NSString stdStringForString:self.name], parameters, scalability_modes);
+      [NSString rtcStdStringForString:self.name], parameters, scalability_modes);
 }
 
 @end
diff --git a/sdk/objc/api/peerconnection/RTCVideoTrack.mm b/sdk/objc/api/peerconnection/RTCVideoTrack.mm
index 5411a22dfd6a2672bd3d8e44e55ff19a8f8ad50a..381feb161766a5d6673cd50b8af03cf0a6997f15 100644
--- a/sdk/objc/api/peerconnection/RTCVideoTrack.mm
+++ b/sdk/objc/api/peerconnection/RTCVideoTrack.mm
@@ -30,7 +30,7 @@
   NSParameterAssert(factory);
   NSParameterAssert(source);
   NSParameterAssert(trackId.length);
-  std::string nativeId = [NSString stdStringForString:trackId];
+  std::string nativeId = [NSString rtcStdStringForString:trackId];
   webrtc::scoped_refptr<webrtc::VideoTrackInterface> track =
       factory.nativeFactory->CreateVideoTrack(source.nativeVideoSource,
                                               nativeId);
diff --git a/sdk/objc/components/video_codec/RTCDefaultVideoEncoderFactory.mm b/sdk/objc/components/video_codec/RTCDefaultVideoEncoderFactory.mm
index 3cc21218045ff4eb5bb87d129a0215baf3b080d6..64e84290b05eed7d6da96901f4b19c3561e83db8 100644
--- a/sdk/objc/components/video_codec/RTCDefaultVideoEncoderFactory.mm
+++ b/sdk/objc/components/video_codec/RTCDefaultVideoEncoderFactory.mm
@@ -91,7 +91,8 @@
       scalabilityMode:(nullable NSString *)scalabilityMode {
   std::optional<webrtc::ScalabilityMode> mode;
   if (scalabilityMode) {
-    mode = webrtc::ScalabilityModeStringToEnum([scalabilityMode stdString]);
+    mode =
+        webrtc::ScalabilityModeStringToEnum([scalabilityMode rtcStdString]);
     if (!mode.has_value()) {
       return [[RTC_OBJC_TYPE(RTCVideoEncoderCodecSupport) alloc]
           initWithSupported:NO];
diff --git a/sdk/objc/components/video_codec/RTCH264ProfileLevelId.mm b/sdk/objc/components/video_codec/RTCH264ProfileLevelId.mm
index 2dc66d1514a409807f145eca846809fba06b587a..a55dc0e3dc8dc43aba0a96b1a0db6822266b1565 100644
--- a/sdk/objc/components/video_codec/RTCH264ProfileLevelId.mm
+++ b/sdk/objc/components/video_codec/RTCH264ProfileLevelId.mm
@@ -45,7 +45,7 @@ NSString *MaxSupportedLevelForProfile(webrtc::H264Profile profile) {
     const std::optional<std::string> profileString = H264ProfileLevelIdToString(
         webrtc::H264ProfileLevelId(profile, profileLevelId->level));
     if (profileString) {
-      return [NSString stringForStdString:*profileString];
+      return [NSString rtcStringForStdString:*profileString];
     }
   }
   return nil;
diff --git a/sdk/objc/helpers/NSString+StdString.h b/sdk/objc/helpers/NSString+StdString.h
index b0324e8a190e2dc8b686039bfde33b2ec42adfc4..0a34e694aadfe78d75e1461a76a26ef15108e547 100644
--- a/sdk/objc/helpers/NSString+StdString.h
+++ b/sdk/objc/helpers/NSString+StdString.h
@@ -18,10 +18,10 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface NSString (StdString)
 
-@property(nonatomic, readonly) std::string stdString;
+@property(nonatomic, readonly) std::string rtcStdString;
 
-+ (std::string)stdStringForString:(NSString *)nsString;
-+ (NSString *)stringForStdString:(const std::string &)stdString;
++ (std::string)rtcStdStringForString:(NSString *)nsString;
++ (NSString *)rtcStringForStdString:(const std::string &)stdString;
 
 @end
 
diff --git a/sdk/objc/helpers/NSString+StdString.mm b/sdk/objc/helpers/NSString+StdString.mm
index 5c2d208b9cfcddb3389ae39f9e018b209aa40e08..93dca0e77a3b6788a4ccc299981edbe2bb69cb94 100644
--- a/sdk/objc/helpers/NSString+StdString.mm
+++ b/sdk/objc/helpers/NSString+StdString.mm
@@ -14,17 +14,17 @@
 
 @implementation NSString (StdString)
 
-- (std::string)stdString {
-  return [NSString stdStringForString:self];
+- (std::string)rtcStdString {
+  return [NSString rtcStdStringForString:self];
 }
 
-+ (std::string)stdStringForString:(NSString *)nsString {
++ (std::string)rtcStdStringForString:(NSString *)nsString {
   NSData *charData = [nsString dataUsingEncoding:NSUTF8StringEncoding];
   return std::string(reinterpret_cast<const char *>(charData.bytes),
                      charData.length);
 }
 
-+ (NSString *)stringForStdString:(const std::string &)stdString {
++ (NSString *)rtcStringForStdString:(const std::string &)stdString {
   // std::string may contain null termination character so we construct
   // using length.
   return [[NSString alloc] initWithBytes:stdString.data()
diff --git a/sdk/objc/native/src/objc_video_decoder_factory.mm b/sdk/objc/native/src/objc_video_decoder_factory.mm
index 8eceb4c4fc390dd14fc59cf35ee8b8b3ebf3da3e..1eac8aa575301ec5c63af63bf84ffbeedcf34802 100644
--- a/sdk/objc/native/src/objc_video_decoder_factory.mm
+++ b/sdk/objc/native/src/objc_video_decoder_factory.mm
@@ -36,7 +36,7 @@ class ObjCVideoDecoder : public VideoDecoder {
  public:
   ObjCVideoDecoder(id<RTC_OBJC_TYPE(RTCVideoDecoder)> decoder)
       : decoder_(decoder),
-        implementation_name_([decoder implementationName].stdString) {}
+        implementation_name_([decoder implementationName].rtcStdString) {}
 
   bool Configure(const Settings &settings) override {
     return [decoder_ startDecodeWithNumberOfCores:settings.number_of_cores()] ==
diff --git a/sdk/objc/native/src/objc_video_encoder_factory.mm b/sdk/objc/native/src/objc_video_encoder_factory.mm
index ba2c5b0c8a1f824499a65cd34fc841b089514072..af0c9f47f1a6643598cd3fc4a63f896fcc738094 100644
--- a/sdk/objc/native/src/objc_video_encoder_factory.mm
+++ b/sdk/objc/native/src/objc_video_encoder_factory.mm
@@ -41,7 +41,7 @@ class ObjCVideoEncoder : public VideoEncoder {
  public:
   ObjCVideoEncoder(id<RTC_OBJC_TYPE(RTCVideoEncoder)> encoder)
       : encoder_(encoder),
-        implementation_name_([encoder implementationName].stdString) {}
+        implementation_name_([encoder implementationName].rtcStdString) {}
 
   int32_t InitEncode(const VideoCodec *codec_settings,
                      const Settings &encoder_settings) override {
diff --git a/sdk/objc/unittests/RTCDataChannelConfigurationTest.mm b/sdk/objc/unittests/RTCDataChannelConfigurationTest.mm
index e2a0ab0e7c6d6f8e64357cc50b7a01cfd29f3968..aeac806937ed926fcbcc8a95e65b8c65db281fc2 100644
--- a/sdk/objc/unittests/RTCDataChannelConfigurationTest.mm
+++ b/sdk/objc/unittests/RTCDataChannelConfigurationTest.mm
@@ -45,7 +45,7 @@
   EXPECT_EQ(maxRetransmits, nativeInit.maxRetransmits);
   EXPECT_EQ(isNegotiated, nativeInit.negotiated);
   EXPECT_EQ(channelId, nativeInit.id);
-  EXPECT_EQ(protocol.stdString, nativeInit.protocol);
+  EXPECT_EQ(protocol.rtcStdString, nativeInit.protocol);
 }
 
 @end
diff --git a/sdk/objc/unittests/RTCIceCandidateTest.mm b/sdk/objc/unittests/RTCIceCandidateTest.mm
index b89a55de6db7ce14e0729c2a729ad2a3cfc23ac2..6eb9d668a1ce25ae3323b1116e4cb17eb1886c5e 100644
--- a/sdk/objc/unittests/RTCIceCandidateTest.mm
+++ b/sdk/objc/unittests/RTCIceCandidateTest.mm
@@ -40,7 +40,7 @@
   EXPECT_EQ(0, nativeCandidate->sdp_mline_index());
 
   std::string sdpString = nativeCandidate->ToString();
-  EXPECT_EQ(sdp.stdString, sdpString);
+  EXPECT_EQ(sdp.rtcStdString, sdpString);
 }
 
 - (void)testInitFromNativeCandidate {
@@ -57,7 +57,7 @@
   EXPECT_TRUE([@"audio" isEqualToString:iceCandidate.sdpMid]);
   EXPECT_EQ(0, iceCandidate.sdpMLineIndex);
 
-  EXPECT_EQ(sdp, iceCandidate.sdp.stdString);
+  EXPECT_EQ(sdp, iceCandidate.sdp.rtcStdString);
 }
 
 @end
diff --git a/sdk/objc/unittests/RTCIceServerTest.mm b/sdk/objc/unittests/RTCIceServerTest.mm
index 0d1919d8bdb1d2495a920cb0ba478c8c05e44477..f84bd6508770016dd369ba2992ebf246eaef508a 100644
--- a/sdk/objc/unittests/RTCIceServerTest.mm
+++ b/sdk/objc/unittests/RTCIceServerTest.mm
@@ -128,10 +128,10 @@
       [[RTC_OBJC_TYPE(RTCIceServer) alloc] initWithNativeServer:nativeServer];
   EXPECT_EQ(1u, iceServer.urlStrings.count);
   EXPECT_EQ("stun:stun.example.net",
-            [NSString stdStringForString:iceServer.urlStrings.firstObject]);
-  EXPECT_EQ("username", [NSString stdStringForString:iceServer.username]);
-  EXPECT_EQ("password", [NSString stdStringForString:iceServer.credential]);
-  EXPECT_EQ("hostname", [NSString stdStringForString:iceServer.hostname]);
+            [NSString rtcStdStringForString:iceServer.urlStrings.firstObject]);
+  EXPECT_EQ("username", [NSString rtcStdStringForString:iceServer.username]);
+  EXPECT_EQ("password", [NSString rtcStdStringForString:iceServer.credential]);
+  EXPECT_EQ("hostname", [NSString rtcStdStringForString:iceServer.hostname]);
   EXPECT_EQ(2u, iceServer.tlsAlpnProtocols.count);
   EXPECT_EQ(2u, iceServer.tlsEllipticCurves.count);
 }
diff --git a/sdk/objc/unittests/RTCMediaConstraintsTest.mm b/sdk/objc/unittests/RTCMediaConstraintsTest.mm
index 7b67147d399d6f621b4f8ab68e0703268d599c33..45d01be07725222ed623f54ae4d83499993e52f7 100644
--- a/sdk/objc/unittests/RTCMediaConstraintsTest.mm
+++ b/sdk/objc/unittests/RTCMediaConstraintsTest.mm
@@ -53,9 +53,9 @@
     NSString *value = [constraints objectForKey:key];
 
     std::string nativeValue;
-    bool found = nativeConstraints.FindFirst(key.stdString, &nativeValue);
+    bool found = nativeConstraints.FindFirst(key.rtcStdString, &nativeValue);
     EXPECT_TRUE(found);
-    EXPECT_EQ(value.stdString, nativeValue);
+    EXPECT_EQ(value.rtcStdString, nativeValue);
   }
 }
 
diff --git a/sdk/objc/unittests/RTCSessionDescriptionTest.mm b/sdk/objc/unittests/RTCSessionDescriptionTest.mm
index 790fcd7bdaf0dbdf13379291be0f70f623c2bf41..9a7c9a22d8e98b0b189bbabb6f06d273436f97c7 100644
--- a/sdk/objc/unittests/RTCSessionDescriptionTest.mm
+++ b/sdk/objc/unittests/RTCSessionDescriptionTest.mm
@@ -44,12 +44,12 @@
 
   std::string sdp;
   nativeDescription->ToString(&sdp);
-  EXPECT_EQ([self sdp].stdString, sdp);
+  EXPECT_EQ([self sdp].rtcStdString, sdp);
 }
 
 - (void)testInitFromNativeSessionDescription {
   const auto nativeDescription = webrtc::CreateSessionDescription(
-      webrtc::SdpType::kAnswer, [self sdp].stdString, nullptr);
+      webrtc::SdpType::kAnswer, [self sdp].rtcStdString, nullptr);
 
   RTC_OBJC_TYPE(RTCSessionDescription) *description =
       [[RTC_OBJC_TYPE(RTCSessionDescription) alloc]
diff --git a/test/ios/test_support.mm b/test/ios/test_support.mm
index aca688c24c9f245e2060c30a184f2cfc77fa5c2b..1f4d1a2c53852830a9c3c5084d8bcf09334dd946 100644
--- a/test/ios/test_support.mm
+++ b/test/ios/test_support.mm
@@ -110,7 +110,7 @@ static std::optional<std::vector<std::string>> g_metrics_to_plot;
 
       exporters.push_back(
           std::make_unique<webrtc::test::ChromePerfDashboardMetricsExporter>(
-              [NSString stdStringForString:outputPath]));
+              [NSString rtcStdStringForString:outputPath]));
     }
   }
   if (!g_webrtc_test_metrics_output_path.empty()) {
@@ -126,7 +126,7 @@ static std::optional<std::vector<std::string>> g_metrics_to_plot;
       exporters.push_back(
           std::make_unique<webrtc::test::MetricsSetProtoFileExporter>(
               webrtc::test::MetricsSetProtoFileExporter::Options(
-                  [NSString stdStringForString:outputPath])));
+                  [NSString rtcStdStringForString:outputPath])));
     }
   }
   webrtc::test::ExportPerfMetric(*webrtc::test::GetGlobalMetricsLogger(),
diff --git a/test/testsupport/ios_file_utils.mm b/test/testsupport/ios_file_utils.mm
index 5c5c7f092a06f4ca9e3de4668c1296b260bbc0ca..06b27c4d59f7abae4a9cb2945bbabdcf2a2401b4 100644
--- a/test/testsupport/ios_file_utils.mm
+++ b/test/testsupport/ios_file_utils.mm
@@ -33,14 +33,14 @@ std::string IOSResourcePath(absl::string_view name,
     // Get full pathname for the resource identified by the name and extension.
     NSString* pathString = [[NSBundle mainBundle] pathForResource:fileName
                                                            ofType:fileType];
-    return [NSString stdStringForString:pathString];
+    return [NSString rtcStdStringForString:pathString];
   }
 }
 
 std::string IOSRootPath() {
   @autoreleasepool {
     NSBundle* mainBundle = [NSBundle mainBundle];
-    return [NSString stdStringForString:mainBundle.bundlePath] + "/";
+    return [NSString rtcStdStringForString:mainBundle.bundlePath] + "/";
   }
 }
 
@@ -51,7 +51,7 @@ std::string IOSOutputPath() {
   @autoreleasepool {
     NSString* tempDir = NSTemporaryDirectory();
     if (tempDir == nil) tempDir = @"/tmp";
-    return [NSString stdStringForString:tempDir];
+    return [NSString rtcStdStringForString:tempDir];
   }
 }
 
